base/ttyhascolor.jl

```diff
@@ -0,0 +1,24 @@
+function ttyhascolor()
+    term_type = get(ENV, "TERM","")
+    startswith(term_type, "xterm") && return true
+    try
+        @static if Sys.KERNEL === :FreeBSD
+            return success(`tput AF 0`)
+        else
+            return success(`tput setaf 0`)
```

Comment:
What happens here on Microsoft Windows?

Reply:
The original code in the repl just returns `true` on windows.

Reply:
It would also be much better not to have two copies of this code; with a few more change it should be possible to remove the one in Terminals.jl.

Reply:
In some circumstances, `tput` may exist on Windows but not all cases (Git Bash Shell, Cygwin, WSL[2]). I suggest that if we catch an exception from `success` meaning that `tput` does not exist then we return `true` if on Windows.

Reply:
I have a candidate to address the redundancy. See 95c8ea3 pending merge to this PR.